### ISSUE TITLE
Exception yielding to phlex component

### ISSUE THREAD

**Issue by joelmoss** [ViewComponent/lookbook#353]
### Describe the bug

Yielding to a Phlex component raises `ActionView::Template::Error: undefined method `div'`

### To Reproduce

```ruby
class HelloComponent < Phlex::HTML
  def template(&)
    div do
      h1 { 'Hello' }
      div(&)
    end
  end
end
```

```ruby
class HelloPreview < Lookbook::Preview
  def default
    render HelloComponent.new do
      div { 'world' }
    end
  end
end
```

### Expected behavior

All works fine when rendering the component in the same way outside a component.

### Version numbers 

Please complete the following information:

* Lookbook: 2.0.0.beta4
* Phlex: 1.5.1
* Rails: 7.0.4.2
* Ruby: 3.2.1

**Comment by allmarkedup** [comment]
Hey @joelmoss, sorry for the slow reply. The problem you are running into is because the component is not actually being rendered in an ActionView context. It's a bit annoying but the only way to work around this is by using the `.div` method on the parent object:

```ruby
class HelloPreview < Lookbook::Preview
  def default
    render HelloComponent.new do |component|
      component.div { 'world' }
    end
  end
end
```

Rendering ViewComponents also has this limitation (even when not using Lookbook). I've not been able to find a better fix for this yet but it should definitely be documented somewhere at a minimum.

Would be nice if the workaround wasn't needed though so I'll try again to find a better solution when I have a bit more time.

If you could let me know if it works for you that would be great :-)

**Comment by joelmoss** [comment]
Yeah that works, thx. Slightly annoying, but I think I can live with it.
